### PR TITLE
collaboration linking: add logging if ambiguous match

### PR DESCRIPTION
**Pull request by MJedr**

### CODE REVIEW

File: inspirehep/modules/workflows/tasks/actions.py [hunk 1]
```diff
@@ -908,6 +908,12 @@ def normalize_collaborations(obj, eng):
             collaboration_normalized_name = response[0].collaboration.value
         LOGGER.info(u"(%s) collaboration normalization: normalized: %s ==> %s", obj.id, collaboration['value'], collaboration_normalized_name)
         if len(response.hits) > 1:
+            matched_collaboration_names = [matched_collaboration.collaboration.value for matched_collaboration in response]
+            LOGGER.info(
+                "(wf: {0}) ambiguous match for collaboration {1}. Matched collaborations: {2}".format(
```
Comment (michamos) [inline inspirehep/modules/workflows/tasks/actions.py:913]: the format string needs to be unicode, not bytes.

File: tests/integration/workflows/fixtures/experiment_with_ambiguous_collaboration_1.json [hunk 2]
```diff
@@ -41,10 +40,10 @@
...
       }
-    ]
-  },
+    ],
   "id": "1109131",
   "links": {
     "json": "http://localhost:5000/api/experiments/1109131?format=json"
-  }
+  },
+  "normalized_name_variants": ["SHIP"]
```
Comment (michamos) [inline tests/integration/workflows/fixtures/experiment_with_ambiguous_collaboration_1.json:48]: why is this changing?

File: inspirehep/modules/workflows/tasks/actions.py [hunk 3]
```diff
@@ -902,12 +902,30 @@ def normalize_collaborations(obj, eng):
...
+            elif len(response.hits) > 1:
+                matched_collaboration_names = [
+                    matched_collaboration.collaboration.value for matched_collaboration in response
+                ]
+                LOGGER.info(
+                    u"(wf: {0}) ambiguous match for collaboration {1}. Matched collaborations: {2}".format(
+                        obj.id, collaboration['value'], ', '.join(matched_collaboration_names)
+                    )
+                )
+                continue
```
Comment (michamos) [inline inspirehep/modules/workflows/tasks/actions.py:914]: this is the branch where you normalize based on a subgroup name, not based on the name of the full collaboration. The flow is a bit complex, but also there it should be unique.

Comment (MJedr) [reply inspirehep/modules/workflows/tasks/actions.py:914]: So it's the wrong place or logging should list subgroup names instead of collaborations?

Comment (michamos) [reply inspirehep/modules/workflows/tasks/actions.py:914]: you should make it clear that it's about collaboration subgroups (`ambiguous match for collaboration subgroup`), and you should also add a check for several results to the non-subgroup matching.

**Review by michamos — APPROVED** [review]

**Pull request merged**